src/TraceEvent/TraceLog.cs

```diff
@@ -1115,6 +1125,11 @@ private unsafe void SetupCallbacks(TraceEventDispatcher rawEvents)
             rawEvents.Clr.GCFinalizeObject += doNothing;
             rawEvents.Clr.MethodJittingStarted += doNothing;
 
+            if (IsRealTime)
```

Comment:
Would be good to add a quick comment here that this is required to ensure that self-describing metadata gets ingested before the event's extended data gets overwritten by the `TraceLog`.

Reply:
👍